command/e2etest/init_test.go

```diff
@@ -375,6 +379,57 @@ func TestInitProviderNotFound(t *testing.T) {
 │ Could not retrieve the list of available versions for provider
 │ hashicorp/nonexist: provider registry registry.terraform.io does not have a
 │ provider named registry.terraform.io/hashicorp/nonexist
+│ 
+│ If you meant to install a provider from the registry that is not in the
```

Comment:
I think for clarity's sake we could drop the first sentence and simply claim `All modules should specify.....`

This could provide an early hedge for a future where even the Hashicorp namespace must be explicitly specified.

Reply:
Good point! This exposes the logic behind when to show the error, but I don't think it's strictly necessary, and it's also (bonus!) shorter to go straight to the "This is what you should be doing" point

Reply:
I've gone ahead and updated it, and updated to make this part of the general suggestion? (the default suggestion if there is no didyoumean from the registry migration matching). Will get to the screenshot, but check it out in the tests!